frontend/src/components/pages/main/Calendar/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { BoxLabel } from "components/common/Box";
+import DateCard from "components/common/DateCard";
+import {
+  CalendarMonthSelectorContainer,
+  CalendarMonthSelectorMonthText,
+  StyledCalendar,
+} from "components/pages/main/Calendar/styles";
+import { addDays, addMonths, setMonth, subDays, subMonths } from "date-fns";
+import getDaysInMonth from "date-fns/getDaysInMonth";
+import lastDayOfMonth from "date-fns/lastDayOfMonth";
+import { Dispatch, HTMLAttributes, SetStateAction, useState } from "react";
+import arrowLeftIcon from "/public/icons/arrow_left.svg";
+import arrowRightIcon from "/public/icons/arrow_right.svg";
+import Image from "next/image";
+
+function populateDateArray(startDate: Date, endDate: Date) {
+  let currentDate = subDays(startDate, 1);
+  const populatedDateArray = [] as Date[];
+  while (currentDate.getDate() !== endDate.getDate()) {
+    populatedDateArray.push(currentDate);
+    currentDate = addDays(currentDate, 1);
+  }
+  populatedDateArray.push(endDate);
+  return populatedDateArray;
+}
+
+function compareUTCYYYYDDMM(date1: Date, date2: Date) {
+  if (date1.getUTCFullYear() === date2.getUTCFullYear()) {
+    if (date1.getUTCMonth() === date2.getUTCMonth()) {
+      if (date1.getUTCDate() === date2.getUTCDate()) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
```

Comment:
개인적으로 depth가 더 적은걸 선호해서 남겨보았습니다 :)

```suggestion
function compareUTCYYYYDDMM(date1: Date, date2: Date) {
  if (date1.getUTCFullYear() === date2.getUTCFullYear()) {
    return false;
  }
  if (date1.getUTCMonth() === date2.getUTCMonth()) {
    return false;
  }
  if (date1.getUTCDate() === date2.getUTCDate()) {
    return false;
  }

  return true;
}
```